File: C#/tres.cs
using System;

class Program 
{
  public static void Main (string[] args) 
  {
    int[,] numero = new int [3,2];

    for(int linha = 0; linha < 3; linha++){
      for(int coluna = 0; coluna < 2; coluna++){
        if (linha == 0 && coluna == 0) {
          Console.WriteLine("Insira um número: ");
          numero[linha, coluna] = Convert.ToInt32(Console.ReadLine());
        } else {
          if(coluna == 0) {
            numero[linha, coluna] = numero[linha - 1, 1] * 2;
          } else {
            numero[linha, coluna] = numero[linha, coluna -1] * 2;
          }
        }
      }
    }
    Console.WriteLine(numero[0,0] + ", " + numero[0,1] + ", " + numero[1,0] + ", " + numero[1,1] + ", " + numero[2,0] + ", " + numero[2,1]);
  }
}

File: JS/tres.js
let numero = []

for(let linha = 0; linha < 2; linha++){
  for(let coluna = 0; coluna < 3; coluna++){ 
    if(linha == 0 && coluna == 0){
      numero[linha, coluna] = Number(prompt("Insira um número: "))
    } else {
      if(coluna == 0) {
        numero[linha, coluna] = numero[linha - 1, 2] * 2
      } else {
        numero[linha, coluna] = numero[linha, coluna - 1] * 2
      }
    }
    console.log(linha)
    console.log(coluna)
    console.log(numero[linha, coluna]);
    //console.log(numero[linha, coluna - 1]) 
  }
}

for(let linha = 0; linha < 2; linha++){
  for(let coluna = 0; coluna < 3; coluna++){ 
   console.log(numero[linha, coluna])
  }
}